core/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -361,15 +359,7 @@ public long parseToMilliseconds(Object value, boolean roundUp,
             } else {
                 strValue = value.toString();
             }
-            return dateParser.parse(strValue, now(context), roundUp, zone);
-        }
-
-        private static Callable<Long> now(QueryRewriteContext context) {
-            return () -> {
-                return context != null
-                        ? context.nowInMillis()
-                        : System.currentTimeMillis();
-            };
+            return dateParser.parse(strValue, context::nowInMillis, roundUp, zone);
```

Comment:
@colings86 I have a test on a branch of mine that reproducibly causes an NPE here because `context` is null. The query is a multi-match query trying to construct a blended term query.

It looks like the previous code handled the null `context`, but this code doesn't. Maybe we need extra null handling here?